packages/flutter_tools/lib/src/commands/packages.dart

```diff
@@ -237,7 +237,7 @@ class PackagesForwardCommand extends FlutterCommand {
   @override
   Future<FlutterCommandResult> runCommand() async {
     Cache.releaseLockEarly();
-    await pub(<String>[_commandName, ...argResults.rest], context: PubContext.pubForward, retry: false);
+    await pubInteractively(<String>[_commandName, ...argResults.rest]);
```

Comment:
Ideally, we'd avoid having both `flutter pub run` and `flutter pub pub run`, but I don't see an easy way to do that. A simple idea which I think may not work is to remove `flutter pub pub`, and delegate to the `PackagesPassthroughCommand` whenever the `PackagesCommand` doesn't recognize the subcommand. This doesn't work because there would be ambiguity between, e.g. the `get` subcommand of `PackagesCommand` and `pub get` (there are others as well).

Reply:
Yeah, I spent a fair amount of time looking at this a few weeks back, and could not make something like that work.

What might work would be to deprecate, and then later remove, all `flutter pub pub` commands entirely. But I think it too soon; we need the `flutter pub` commands out in stable for a few releases before I would feel safe doing that.